docs/source/indexes/cagra.rst

```diff
@@ -98,53 +98,148 @@ Tuning Considerations
 The 3 hyper-parameters that are most often tuned are `graph_degree`, `intermediate_graph_degree`, and `itopk_size`.
 
 Memory footprint
-----------------
+================
 
-CAGRA builds a graph that ultimately ends up on the host while it needs to keep the original dataset around (can be on host or device).
+CAGRA stores the dataset (raw vectors) and nearest-neighbor graph (neighbor IDs) in memory.
+The dataset can be on **host or device**; the graph is **pinned to host memory**.
 
-IVFPQ or NN-DESCENT can be used to build the graph (additions to the peak memory usage calculated as in the respective build algo above).
+Baseline Memory Footprint
+--------------
 
-Dataset on device (graph on host):
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+These are the steady-state sizes of the stored data structures.
 
-Index memory footprint (device): :math:`n\_index\_vectors * n\_dims * sizeof(T)`
+.. math::
+
+   \text{dataset\_size}
+   \;=\;
+   \text{number\_vectors} \times \text{vector\_dimension} \times \text{bytes\_per\_dimension}
+
+.. math::
+
+   \text{graph\_size (host)}
+   \;=\;
+   \text{number\_vectors} \times \text{graph\_degree} \times \operatorname{sizeof}\!\big(\mathrm{IdxT}\big)
+
+**Example** (1,000,000 vectors, dim = 1024, fp32, graph\_degree = 64, IdxT = int32):
+
+- dataset\_size = 4,096,000,000 B = **3906.25 MiB**
+- graph\_size   = 256,000,000 B = **244.14 MiB**
+
+Build peak memory usage
+-----------------------
+
+Index build has two phases: (1) construct an intermediate graph, then (2) optimize it (prune/reorder).
+The initial graph can be built with IVF-PQ (supports out-of-core, so datasets larger than GPU memory can be streamed)
+or NN-descent. The steps below are sequential with distinct peak memory consumption. The overall peak memory utilization depends on the configured memory resource.
+
+Out-of-core IVF-PQ
+~~~~~~~~~~~~~~~~~~~~~~~
+
+**IVF-PQ Build (centroid training)** — uses a training subset to compute cluster centroids and assignments.
```

Comment:
Maybe briefly summarize the way these steps are used together to scale the construction of the knn graph? Another thing we can (and should) add is the use of the all-neighbors graph API to construct the KNN graph in a super scalable way and then calling `cagra::optimize()` on the knn graph to yield the final CAGRA graph. That should require only a few lines of code and would allow a cagra graph to be built super fast. Further, we should update this to include enabling the new ACE algorithm. Not sure if we should include the CAGRA iterative build yet, as that hasn't been deep tested quite yet.